Bottom: b9ae45adc178b051acd2f409592189c7637925f1
Top:    10f4c57e14f5c5c3361d6c4e41a2ab34af29ed13
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2012-11-21 02:45:05 +0000

PCB+GL: Fix GL headers for Win32


---

diff --git a/src/hid/common/hidgl.c b/src/hid/common/hidgl.c
index 8170fc3..f7287ce 100644
--- a/src/hid/common/hidgl.c
+++ b/src/hid/common/hidgl.c
@@ -32,12 +32,18 @@
 #include <math.h>
 #include <assert.h>
 
-/* The Linux OpenGL ABI 1.0 spec requires that we define
- * GL_GLEXT_PROTOTYPES before including gl.h or glx.h for extensions
- * in order to get prototypes:
- *   http://www.opengl.org/registry/ABI/
- */
-#define GL_GLEXT_PROTOTYPES 1
+#ifdef WIN32
+#   define WIN32_LEAN_AND_MEAN 1
+#   include <windows.h>
+#else
+    /* The Linux OpenGL ABI 1.0 spec requires that we define
+     * GL_GLEXT_PROTOTYPES before including gl.h or glx.h for extensions
+     * in order to get prototypes:
+     *   http://www.opengl.org/registry/ABI/
+     */
+#   define GL_GLEXT_PROTOTYPES 1
+#endif
+
 #ifdef HAVE_OPENGL_GL_H
 #   include <OpenGL/gl.h>
 #else
@@ -67,7 +73,6 @@
 #include <dmalloc.h>
 #endif
 
-
 triangle_buffer buffer;
 float global_depth = 0;
